reVX/plexos/base.py

```diff
@@ -561,3 +586,140 @@ def _ammend_output_meta(self, row_idx, sc_gids, res_gids, gen_gids,
             self._output_meta.at[index, 'res_gids'] += res_gids
             self._output_meta.at[index, 'gen_gids'] += gen_gids
             self._output_meta.at[index, 'res_built'] += res_built
+
+    @staticmethod
+    def tz_convert_profiles(profiles, timezone):
+        """Convert profiles to local time and forward/back fill missing data.
+
+        Parameters
+        ----------
+        profiles : np.ndarray
+            Profiles of shape (time, n_plants) in UTC
+        timezone : str
+            Timezone for output generation profiles. This is a string that will
+            be passed to pytz.timezone() e.g. US/Pacific, US/Mountain,
+            US/Central, US/Eastern, or UTC. For a list of all available
+            timezones, see pytz.all_timezones
+
+        Returns
+        -------
+        profiles : np.ndarray
+            Profiles of shape (time, n_plants) in timezone
+        """
+
+        logger.info('Converting profiles timezone to {}'.format(timezone))
+
+        if len(profiles) < 8760:
+            msg = ('Cannot use profiles that are not at least hourly! '
+                   'Received shape {}'.format(profiles.shape))
+            logger.error(msg)
+            raise RuntimeError(msg)
+
+        steps_per_hour = len(profiles) // 8760
+
+        # use jan 1 to avoid daylight savings
+        date = datetime.datetime(2011, 1, 1)
+        date = pytz.timezone(timezone).localize(date)
+        tz_offset = int(date.strftime('%z')[:3])
+        roll_int = steps_per_hour * tz_offset
+
+        profiles = np.roll(profiles, roll_int, axis=0)
+
+        if roll_int < 0:
+            for i in range(roll_int, 0):
+                # don't fill nighttime for solar
+                if not (profiles[i, :] == 0).all():
+                    profiles[i, :] = np.nan
+            profiles = pd.DataFrame(profiles).ffill().values
+        elif roll_int > 0:
+            for i in range(1, roll_int + 1):
+                # don't fill nighttime for solar
+                if not (profiles[i, :] == 0).all():
+                    profiles[i, :] = np.nan
+            profiles = pd.DataFrame(profiles).bfill().values
+
+        return profiles
+
+    @staticmethod
+    def get_unique_plant_names(table, name_col, tech_tag=None):
+        """Get a list of ordered unique plant names
+
+        Returns
+        -------
+        list | None
+        """
+        names = None
+        if name_col is not None:
+            if name_col not in table:
+                msg = ('Could not find requested name column "{}" in plexos '
+                       'table, the available columns are: {}'
+                       .format(name_col, sorted(table.columns.values)))
+                logger.error(msg)
+                raise KeyError(msg)
+
+            names = table[name_col].values.tolist()
+
+            if tech_tag is not None:
+                names = [name + f' {tech_tag}' for name in names]
+
+            seen = set()

```

Comment:
You could consider using `Counter` here:

```
from collections import Counter
counter = Counter()
for i, name in enumerate(names):
    counter.update([name])
    count = counter[name]
    if count > 1:
        names[i] = name + " {}".format(count)
```

Reply:
Cool i didnt know about counter! I implemented. Same number of lines i think but more robust and easy to read.